code/datums/preferences.dm

```diff
@@ -249,6 +279,41 @@ datum/preferences
 
 		user << browse(dat,"window=preferences;size=333x615")
 
+
+	//id, The name of the Select table ID to be used.
+	//ah_var, The var in the appearance holder that is in focus
+	//Style_list, The assoc list with the values to be used for generating this select table
+	proc/generate_select_table(var/id, var/ah_var, var/list/style_list)
+		var/select = "<Select id='[id]'>"
+		for (var/i in style_list)
+			//this is for setting the default value
+			if (AH && i == ah_var)
+				select += "<option value='[style_list[i]]' selected='selected'>[i]</option>"
+			else 
+				select += "<option value='[style_list[i]]'>[i]</option>"
+		select += "</Select>"
+		return select
+
+	Topic(href, href_list[])
+		var/table_id = href_list["id"]
+
+		//the if block determines where to save the incoming choice in the AppearanceHolder. Then in all cases it will update the icon and send it to the browser
+		if (table_id)
+			if (table_id == "underwear")
+				AH.underwear = href_list["style"]
+			else if (table_id == "custom_first")
+				AH.customization_first = href_list["style"]
+			else if (table_id == "custom_second")
+				AH.customization_second = href_list["style"]
+			else if (table_id == "custom_third")
+				AH.customization_third = href_list["style"]
+
+			update_preview_icon()

```

Comment:
Possibly sanity check here to ensure the requested preview icon isn't identical to the existing preview icon, to avoid extraneous update_preview_icon calls in edge cases.

Reply:
Though this should never occur without someone screwing with http requests, I now make sure that one of the values sent is different from current value in the appearanceHolder.
https://github.com/goonstation/goonstation-2016/pull/90/commits/a1892c37f7f36f7fd0fa88a639d14243347f4e18